lib/src/native/realm_core.dart

```diff
@@ -2645,6 +2712,65 @@ extension LevelExt on Level {
   }
 }
 
+class WriteCompleter with Cancellable implements Completer<void> {
+  final Completer<void> _internalCompleter;
+  final CancellationToken? _cancellationToken;
+  int? _id;
+  final Realm _realm;
+
+  WriteCompleter(this._realm, this._cancellationToken) : _internalCompleter = Completer<void>() {
+    final ct = _cancellationToken;
```

Comment:
To allow dart null analysis to kick in after the `if (ct != null)` check. Otherwise, we'll have to always add `!` when calling API on `_cancellationToken`.

Reply:
got it

Reply:
You can try, it won't. It's a class level variable, so it can't ensure that it's not set to null after the if check.

Reply:
got it.